x-pack/plugins/ingest_manager/server/services/epm/packages/get.ts

```diff
@@ -21,32 +26,51 @@ export async function getCategories(options: Registry.CategoriesParams) {
   return Registry.fetchCategories(options);
 }
 
+// TODO this function is odd to me, it will use `fetchList` to get the **latest** packages and then squash them together
```

Comment:
@jfsiii The reason I needed to modify this function is that when this code block is called: https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/epm/kibana/index_pattern/install.ts#L83-L96

from here: https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/epm/packages/install.ts#L375

It will attempt to retrieve the _**latest**_ package information using the `/search` api of the registry even if the currently installed version isn't the latest. I'm not sure if that's intentional.

Should this: https://github.com/elastic/kibana/blob/master/x-pack/plugins/ingest_manager/server/services/epm/kibana/index_pattern/install.ts#L94

be caching the latest package information or the currently installed package information?

If this call has to get the _**latest**_ package, then we'll fail here and block the UI in the scenario I'm considering. which is an issue.

Basically, the `installIndexPatterns` function will get called for `system` for example, and it will get all the packages saved objects information (image the endpoint package is on 0.16.0), it'll then search for the latest package from the registry for all installed packages (endpoint will show `0.17.0`) it will then attempt to download the latest package information for each installed package which will fail in the CDN issue or in my hacked scenario when it tries to download the `0.17.0-dev.0` package even though this function really only cares about installing the index patterns for the `system` package.

Reply:
cc @ruflin & @neptunian for history & thoughts on logic for latest vs requested/installed